xdsl/transforms/arith_add_fastmath.py

```diff
@@ -0,0 +1,90 @@
+"""Passes to manipulate fastmath flags in FP arith operations."""
+
+from dataclasses import dataclass
+from typing import Literal
+
+from xdsl.dialects import arith, builtin, llvm
+from xdsl.ir import Operation
+from xdsl.passes import MLContext, ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+)
+
+
+def _match_string_to_enum(flag: str, enum_type: type[llvm.FastMathFlag]):
+    for member in enum_type:
+        if flag == str(member.value):
+            return member
+
+    raise ValueError(f"{flag} is not a valid fastmath flag.")
+
+
+def _get_flag_list(flags: list[str], enum_type: type[llvm.FastMathFlag]):
+    list_flags: list[enum_type] = []
+    for flag in flags:
+        list_flags.append(_match_string_to_enum(flag, llvm.FastMathFlag))
+    return list_flags
+
+
+@dataclass
+class AddArithFastMathFlags(RewritePattern):
+    """Adds fastmath flags to FP binary operations from arith dialect."""
+
+    arith_op_cls: type[arith.FloatingPointLikeBinaryOp]
+    fastmath_op_attr: arith.FastMathFlagsAttr
+
+    def match_and_rewrite(self, op: Operation, rewriter: PatternRewriter) -> None:
```

Comment:
Wouldn't
```suggestion
    @op_type_rewrite_pattern
    def match_and_rewrite(self, op: arith.Addf | arith.Subf | arith.Mulf | arith.Divf | arith.Minimumf | arith.Maximumf, rewriter: PatternRewriter) -> None:
```
achieve what you hardcode at the Pass level either way? Now, maybe I'm missing a rationale to not use it